packages/@aws-cdk/service-spec-types/src/types/resource.ts

```diff
@@ -496,6 +497,34 @@ export interface RelationshipRef {
   readonly propertyName: string;
 }
 
+/**
+ * Represents the types of logs a Cloudformation Resource can produce and what destinations can consume them
+ */
+export interface VendedLog {
+  /**
+   * What version of permissions the destination supports V1 | V2
+   */
+  permissionsVersion: string;
+  /**
+   * List of the types of logs a Cloudformation resource can produce
+   */
+  readonly logTypes: string[];
+  /**
+   * List of the destinations the can consume those logs
+   */
+  readonly logDestinations: LogDestination[];
+}
+
+/**
+ * Represents a destination that logs from a Cloudformation resource can be sent to
+ */
+export interface LogDestination {
+  /**
+   * Resource type of the destination S3 | CWL | FH | XRAY
+   */
+  readonly destinationType: string;
```

Comment:
There should be no reason to not strongly type this:

```suggestion
  readonly destinationType: "S3" | "CWL" | "FH" | "XRAY";
```